### PR TITLE
Book dto

### PR DESCRIPTION
**Pull request by Nikname2303**

### CODE REVIEW

File: src/main/java/com/example/bookshop/dto/BookDto.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.example.bookshop.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
```
Comment (JohnGrey17) [inline src/main/java/com/example/bookshop/dto/BookDto.java:5]: maybe will be better to use separate annotation as @Getter @Setter @ToString ect

File: src/main/java/com/example/bookshop/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.example.bookshop.dto;
+
+import java.math.BigDecimal;
+import lombok.Data;
+
```
Comment (JohnGrey17) [inline src/main/java/com/example/bookshop/dto/CreateBookRequestDto.java:5]: same here

**Review by JohnGrey17 — COMMENTED** [review]
Looks good for me added minor suggestion

File: pom.xml [hunk 3]
```diff
@@ -41,11 +43,25 @@
...
 		</dependency>
+		<dependency>
+			<groupId>org.mapstruct</groupId>
+			<artifactId>mapstruct</artifactId>
+			<version>${mapstruct.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
+			<version>1.18.32</version>
```
Comment (andrii-hoienko) [inline pom.xml]: Better to create properties for all versions

File: src/main/java/com/example/bookshop/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/books")
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookshop/controller/BookController.java]:
```suggestion
@RequestMapping("api/books")
```

File: src/main/java/com/example/bookshop/dto/BookDto.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+@Data
+public class BookDto {
+    private Long id;
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookshop/dto/BookDto.java]:
```suggestion
```

File: src/main/java/com/example/bookshop/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -51,4 +48,12 @@ public List<Book> findAll() {
             throw new DataProcessingException("Can't get books ", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookshop/repository/impl/BookRepositoryImpl.java:55]: Please add catch

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good job, but some improvements required

File: src/main/java/com/example/bookshop/repository/impl/BookRepositoryImpl.java [hunk 7]
```diff
@@ -51,4 +49,14 @@ public List<Book> findAll() {
...
+            Book book = entityManager.find(Book.class, id);
+            return Optional.ofNullable(book);
+        } catch (RuntimeException e) {
```
Comment (karma-o) [inline src/main/java/com/example/bookshop/repository/impl/BookRepositoryImpl.java]:
```suggestion
        } catch (Exception e) {
```
Just to be sure we won't miss anything

File: pom.xml [hunk 8]
```diff
@@ -41,11 +43,24 @@
...
 		</dependency>
+		<dependency>
+			<groupId>org.mapstruct</groupId>
+			<artifactId>mapstruct</artifactId>
+			<version>${mapstruct.version}</version>
+		</dependency>
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
+		</dependency>
```
Comment (karma-o) [inline pom.xml:64]: why did you remove the specified version? Andrii said that we should just move it to property(variable) as you did with other dependencies like mapstruct

**Pull request merged**